CONVENTIONS.md

```diff
@@ -116,12 +116,72 @@ Name screen coordinates `x` and `y`, rather than `left` and `top`.
+
+Javadoc will take the first sentence, ended by the first `.`, as a brief description of the member you are documenting; you can use `{@summary }` inline tag to enclose another sentence to serve as the description, such as when you have `.` within sentences like `{@summary Returns the client's player, i.e. the controlled player.}`.
+
+### Tooling
+
+Fabric's official hosted javadocs are generated with [Java 11's standard doclet](https://docs.oracle.com/en/java/javase/11/docs/specs/doc-comment-spec.html) and can use any feature it supports, though you can personally build the documentation with newer Java. See ["checking javadoc"](#checking-javadoc) for how to build the documentation locally.
+
+### Custom tags
+
+A few additional block tags are supported:
+ - `@apiNote`: API Notes. A few comments for users of the documented member.
+ - `@implSpec`: Implementation Specification. Tells how this method should be implemented.
+ - `@implNote`: Implementation Notes. A few comments on the implementation.
+
+Feel free to use these tags and write under these sections.
+
+### Linking to other members
 
 Use `@link`, `@linkplain` and `@see` tags to refer to other parts of the code.
 
+When linking to another class, you can use the simple name of a class if it's from `java.lang` package or from the same package, or it appears in the signature of the class or its members (methods or fields).
+
+<details>
+<summary>
+An example
+</summary>
+
+Assume this is the decompiled content in enigma, which does not show imports:
+```java
+/**
+ * Assume this class is from the {@code net.example.stuff} package.
+ *
+ * <p>You can link to {@link Optional} as it's part of the class signature (type parameter bound).
+ *
+ * <p>You must fully qualify {@link net.example.stuff.basic.BasicStuffUser} when linking as it is not in
+ * any signature and is from a different package.
+ */
+public class Stuff<T extends Optional<?>> {
+  /**
+   * You can link to {@link Listener} with the simple name as it's part of a field's signature.
+   */
+  protected Listener listener;
+
+  /**
+   * You can link to {@link List} with the simple name as it's part of a method's signature.
+   *
+   * <p>You must fully qualify {@link net.example.util.UtilityClass} when linking because it is not part
+   * of any signature (even though it is used in code) and is from a different package.
+   */
+  public Stuff(List<Integer> opt) {
+    UtilityClass.callMethod(opt);
+  }
+}
+```
+</details>
+
+### Checking javadoc
+After you wrote your javadoc, you should check its validity, such as looking at the `Run ./gradlew build javadocJar checkMappings mapNamedJar --stacktrace` section of your GitHub pull request build result (for any jdk). Look at the `Task :javadoc` section.
```

Comment:
```suggestion
After writing a Javadoc, you should check its validity. This can easily be done from GitHub using the 'Checks' section. For any JDK version, the section titled `Run ./gradlew build javadocJar checkMappings mapNamedJar --stacktrace` will contain output related to Javadoc generation in the `Task :javadoc` section.
```